src/bin/busd.rs

```diff
@@ -44,7 +57,21 @@ async fn main() -> Result<()> {
 
     let args = Args::parse();
 
-    let mut bus = bus::Bus::for_address(args.address.as_deref()).await?;
+    // TODO: when we have `Config` from #159, prefer `config.r#type` before `BusType::default()`
+    let bus_type = if args.system {
+        BusType::System
+    } else {
+        BusType::default()
+    };
+
+    // TODO: when we have `Config` from #159, prefer `config.listen` before `try_from(bus_type)`
+    let address = if let Some(address) = args.address {
+        Address::try_from(address.as_str())?
+    } else {
+        Address::try_from(bus_type)?
+    };
+
+    let mut bus = bus::Bus::for_address(&format!("{address}")).await?;
```

Comment:
Note that I'm a little anxious about all the back and forth between `Address` and `&str` / `String` we have here now

It only happens during start up, so I'm not sure whether it's worth trying to optimise this now, or do this holistically once `Config` and other functionality is working